### ISSUE TITLE
`terraform_unused_required_providers` can't handle local variables + for each

### ISSUE THREAD

**Issue by hluaces** [terraform-linters/tflint#1455]
## Introduction

Tflint `0.39.0` broke the `terraform_unused_required_providers` rule when using `locals` and `for_each` on a resource.

There's a MRE here:

- https://github.com/hluaces/tflint-0.39-bug

## Expected Behavior

The MRE should not raise an error. It used to work on `0.38.1`:

```console
$ git remote -v
origin	git@github.com:hluaces/tflint-0.39-bug.git (fetch)
origin	git@github.com:hluaces/tflint-0.39-bug.git (push)

$ tflint-0.38.1 --version
TFLint version 0.38.1

$ tflint-0.38.1 --init

$ tflint-0.38.1 

$ echo $?
0
```

## Actual behavior

```console
$ git remote -v
origin	git@github.com:hluaces/tflint-0.39-bug.git (fetch)
origin	git@github.com:hluaces/tflint-0.39-bug.git (push)

$ tflint --init

$ tflint
1 issue(s) found:

Warning: provider 'datadog' is declared in required_providers but not used by the module (terraform_unused_required_providers)

  on versions.tf line 4:
   4:     datadog = {
   5:       source  = "Datadog/datadog"
   6:       version = "~> 3.0"
   7:     }

Reference: https://github.com/terraform-linters/tflint/blob/v0.39.0/docs/rules/terraform_unused_required_providers.md
```

## Step to Reproduce

```console
$ git clone git@github.com:hluaces/tflint-0.39-bug.git 

$ cd tflint-0.39-bug/

$ tflint --init

$ tflint

```

## Additional Context

Only on `0.39.0`, `0.38.1` works fine.
